src/main/java/se/sundsvall/chat/integration/db/model/MessageEntity.java

```diff
@@ -0,0 +1,214 @@
+package se.sundsvall.chat.integration.db.model;
+
+import static java.time.OffsetDateTime.now;
+import static java.time.temporal.ChronoUnit.MILLIS;
+
+import jakarta.persistence.CascadeType;
+import jakarta.persistence.CollectionTable;
+import jakarta.persistence.Column;
+import jakarta.persistence.ElementCollection;
+import jakarta.persistence.Entity;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.ForeignKey;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.OneToMany;
+import jakarta.persistence.PrePersist;
+import jakarta.persistence.Table;
+import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.util.List;
+import java.util.Objects;
+import org.hibernate.annotations.UuidGenerator;
+
+@Entity
+@Table(name = "message")
+public class MessageEntity {
+
+	@Id
+	@UuidGenerator
+	@Column(name = "id")
+	private String id;
+
+	@Column(name = "sequence_number")
```

Comment:
Could we use a sequence here?

```
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "mySeqGen")
    @SequenceGenerator(name = "mySeqGen", sequenceName = "my_sequence_name", allocationSize = 1)
```

Reply:
That wont work. From the doc on `@GeneratedValue`

```
Provides for the specification of generation strategies for the values of primary keys.
The GeneratedValue annotation may be applied to a primary key property or field of an entity or mapped superclass in conjunction with the Id annotation. The use of the GeneratedValue annotation is only required to be supported for simple primary keys. Use of the GeneratedValue annotation is not supported for derived primary keys.

```

Unless we want the sequence_number to be the ID of the entity. But I dont think we want that.

Reply:
I took the errandNumberSequence logic from SM instead. That also means we get one sequence per namespace+municipalityId. And if we want we could add conversationId to it without much effort.